kusto/ingest/internal/resources/resources.go

```diff
@@ -152,6 +154,7 @@ func New(client mgmter) (*Manager, error) {
 	}
 
 	m.kustoTokenCacheExpiration = time.Now().UTC()
+	m.lastFetchTime.Store(time.Now().UTC())
 	go m.renewResources()
```

Comment:
This API eventually uses m.fetchLock.
it's best to move it past line 161

Reply:
Line 161 does not acquire the lock, just initializes the lock object.
Either way - I removed it.

Reply:
I meant you run a go routine using the lock object **before** you initialize it.
This seems prone to race conditions.